src/tracks/Pattern.cpp

```diff
@@ -65,6 +65,39 @@ Pattern::Pattern( InstrumentTrack * _instrument_track ) :
 	m_steps( MidiTime::stepsPerTact() )
 {
 	setName( _instrument_track->name() );
+
+	// Resize this track to be the same as existing tracks in the BB
+	// Only if there already exists at least another track in the BB
+	const TrackContainer::TrackList & tracks = m_instrumentTrack->trackContainer()->tracks();
+	if(tracks.size() > 1)
+	{
+		// Get currentTCO index (current BB)
+		int currentTCO = m_instrumentTrack->getTCOs().indexOf(this);
+		// Get first InstrumentTrack
+		unsigned int trackID = 0;
+		for(; trackID < tracks.size(); ++trackID)
+		{
+			// skip track that is the same as the current Pattern
+			if(tracks.at(trackID) == static_cast<Track *>(m_instrumentTrack))
```

Comment:
When you hit this condition, there are no more instrument tracks. You should simply check that the track is an instrument track; then if the track is different from `m_instrumentTrack`, `m_steps` is updated.